Tools-Override/tests.targets

```diff
@@ -62,8 +65,60 @@
 
   <!-- The Code Coverage targets will override TestHost and TestCommandLine if coverage is enabled -->
   <Import Project="$(MSBuildThisFileDirectory)CodeCoverage.targets" />
+  
+  <!-- This will generate a version of the dotnet cli that runs with the live
+       build of corefx using the other components gathered from the stable 
+       version consumed from the Tools directory
+       To debug any issues set COREHOST_TRACE=1 and rerun dotnet
+  -->
+  <UsingTask TaskName="GenerateDepsJson" AssemblyFile="$(BuildToolsTaskDir)CoreFx.Tools.dll"/>  
+  <Target Name="ConstructTestingHost" Condition="!Exists('$(TestSharedFxDir)')" >
```

Comment:
What happens if something in this target fails in some way? The TestSharedFxDir will be present but this target will not run any longer. Will people have to do a full clean to fix this? For that reason I think it might be better to use a sempahore file that gets written at the end.